src/components/DisplayMetrics.js

```diff
@@ -15,9 +16,12 @@ class DisplayMetrics extends React.Component {
   componentWillReceiveProps(props) {
     props.dataCSV.data.forEach((element, i) => {
       this.setState((state, props) => ({
-        suggestions: [
-          ...state.suggestions,
-          { label: props.dataCSV.data[i].campaign },
+        suggestionsForCampaign: [
+          ...state.suggestionsForCampaign,
+          { label: props.dataCSV.data[i].campaign }
+        ],
+        suggestionsForChannel: [
+          ...state.suggestionsForChannel,
           { label: props.dataCSV.data[i].channel }
```

Comment:
DRY 

i would also place this into a utils function, to separate business logic from the view